### PR TITLE
preset the raster color input with categories` values

### PR DESCRIPTION
**Pull request by apercas**

### CODE REVIEW

**Comment by simaob** [comment]
hi @apercas what's the status of this?

**Comment by apercas** [comment]
@simaob I'll push it before lunchtime

**Comment by apercas** [comment]
@simaob Sorry it took more than expected, I've been distracted with other projects, please give it a try.

**Comment by simaob** [comment]
The modal looks good, but when I see it is not filled with the current selected values.

**Comment by simaob** [comment]
I still see some issues, different things in different browsers:

**Firefox**
- When I first open a page to edit a raster page and click _Import/Export Raster Colors_ I get the modal box, but the values are not filled

**Chrome**
- in Chrome the values are pre-filled, but if I change the colours and click again on the _Import/Export Raster Colors_ it doesn't change the "pre-filled"

Also if I change the colors inside the model and click "Back" it doesn't apply any changes that I've made inside the modal.

**Comment by apercas** [comment]
Checking

File: app/assets/javascripts/views/map_file_categories_view.js [hunk 1]
```diff
@@ -30,6 +34,17 @@
       $(this.featherRaster).featherlight();
     },
 
+    clickFeatherRaster: function() {
+      var self = this;
+      var saveFormBtn = document.getElementsByClassName('-saveform')[0];
+      saveFormBtn.style.display = 'block';
+      saveFormBtn.onclick = function() {
+        self.rasterColorInput.value = document.getElementsByClassName('raster_color_input')[1].value;
+        debugger
```
Comment (simaob) [inline app/assets/javascripts/views/map_file_categories_view.js]: gremlin alert!

**Comment by simaob** [comment]
It's getting there. =) I can see the modal populated, and I can make changes to the content of the modal and save, and that works. Is it doing a new request? Would it be possible to just apply it to the pickers?

But on Firefox if I change on the colour pickers it doesn't update the content of the modal for some reason:(

File: app/assets/javascripts/views/map_file_categories_view.js (hunk 1 above)
Comment (apercas) [inline app/assets/javascripts/views/map_file_categories_view.js]: Damn, I missed that because of a hurry! Cleaned!

**Comment by apercas** [comment]
As there is no relation between each single line and each column we should loop over them in each save, this implies (if I'm not mistaken) rebuilding the categories map every time. Doing it on the reload we do it that just once (because it's necessary). But I can be mistaken, though!

I've pushed a hack for the Firefox bug as well.

**Comment by simaob** [comment]
It seems to be working now.

**Pull request merged**